script/upgrades/MINIDROP/data/generate-airdrop-csv.ts

```diff
@@ -0,0 +1,97 @@
+// Each execution of this will consume the following amount of credits:
+// Formula: Credits = Datapoints / Datapoints per credit
+// since we are on the premium plan, we have 25.000 datapoints per credit
+// The Datapoints are calculated by the number of rows * the number of columns
+// In our case, as of writing this, we have 3.648.047 rows and 5 columns
+// Meaning per execution we consume: (3.648.047 * 5) / 25.000 = 729,6 credits
+
+import { config } from "dotenv";
+import { createObjectCsvWriter } from "csv-writer";
+import Decimal from "decimal.js";
+import fs from "fs";
+import path from "path";
+
+interface Airdrop {
+  address: string;
+  mentoAllocation: string;
+  cUSDAllocation: string;
+}
+
+config();
+
+async function run() {
+  const duneApiKey = process.env.DUNE_API_KEY;
+  const filepath = path.resolve(__dirname, "airdrop.csv");
+  clearFileContents(filepath);
+
+  const csvWriter = createObjectCsvWriter({
+    path: filepath,
+    header: [
+      { id: "address", title: "Address" },
+      { id: "mentoAllocation", title: "MENTO Allocation" },
+      { id: "cUSDAllocation", title: "cUSD Allocation" },
+    ],
+    append: false,
+  });
+
+  const queryId = 3932204; // https://dune.com/queries/3932204
+
+  if (!duneApiKey) {
+    console.error("🚨 Missing DUNE_API_KEY environment variable");
+    process.exit(1);
+  }
+
+  const options = { method: "GET", headers: { "X-DUNE-API-KEY": duneApiKey } };
+  const limit = 10000;
+  let uri = `https://api.dune.com/api/v1/query/${queryId}/results?limit=${limit}`;
+
+  let hasMorePages = true;
+  let rowsFetched = 0;
+  while (hasMorePages) {
+    let jsonResponse;
+    try {
+      const response = await fetch(uri, options);
+      jsonResponse = await response.json();
+      rowsFetched += jsonResponse.result.rows.length;
+      console.log(`⏳ ${rowsFetched}/${jsonResponse.result.metadata.total_row_count} records fetched`);
+    } catch (err) {
+      console.error(err);
```

Comment:
If the API call fails this block will continue the execution, is that a problem here?

Reply:
I would say you are right a throw would be nicer here 👍